.github/workflows/lint-test-chart.yaml

```
@@ -18,6 +18,28 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Install Helm
+        uses: azure/setup-helm@fe7b79cd5ee1e45176fcad797de68ecaf3ca4814 # v4.2.0
+        with:
+          token: ${{ github.token }}
+          version: latest
+
+      - name: Run helm values.schema.json check
+        working-directory: charts/external-dns
+        run: |
+          set -euo pipefail
+          helm plugin install https://github.com/losisin/helm-values-schema-json.git
+
+          helm schema -output diff-schema.schema.json
+
+          CURRENT_SCHEMA=$(cat values.schema.json)
+          GENERATED_SCHEMA=$(cat diff-schema.schema.json)
+          if [ "$CURRENT_SCHEMA" != "$GENERATED_SCHEMA" ]; then
+            echo "Schema not up to date. Please run 'scripts/helm-tools.sh --schema' and commit changes!" 1>&2
+            diff diff-schema.schema.json values.schema.json
+            exit 1
+          fi
```

```suggestion
      - name: Run Helm Schema  check
        working-directory: charts/external-dns
        run: |
          set -euo pipefail
          
          helm plugin install https://github.com/losisin/helm-values-schema-json.git

          helm schema

          if [[ -n "$(git status --porcelain --untracked-files=no)" ]]
          then
            echo "Schema not up to date. Please run helm schema and commit changes!" >&2
            exit 1
          fi
```
I think this pattern would be clearer and matches the pattern used for `helm-docs`.